drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/commands/actualcommands/AbstractScenarioSimulationCommand.java

```diff
@@ -42,13 +44,65 @@
  */
 public abstract class AbstractScenarioSimulationCommand extends AbstractCommand<ScenarioSimulationContext, ScenarioSimulationViolation> {
 
+    private static final AtomicLong COUNTER_ID = new AtomicLong();
+
+    /**
+     * Auto-generated incremental identifier used  to uniquely identify each command
+     */
+    private final long id;
+
+    /**
+     * Flag that indicates if the command is <b>undoable</b>. Default is <code>false</code>
+     */
+    private final boolean undoable;
+
+    /**
+     * The <code>ScenarioSimulationContext.Status</code> to restore when calling <b>undo/redo</b>.
+     * Needed only for <b>undoable</b> commands.
+     */
+    protected ScenarioSimulationContext.Status restorableStatus = null;
+
+
+    /**
+     * Calling this constructor will set the command as <b>undoable</b>
+     *
+     *
+     * @param undoable
+     */
+    protected AbstractScenarioSimulationCommand(boolean undoable) {
```

Comment:
I think it could be useful to have also the default constructor with undoable = `true` (or `false` if you think that this should be the default) so it is not needed to call `super(true/false)` in all the implementations

Reply:
I would prefer that actual command must esplicitly declare if they are undoable or not (and so the developer) without giving a default choice, to enforce the requirement.
Giving a "default"  would only open the way for possible mistake without giving a real advantage (the default constructor should be anyway protected - if made public, the possible source of mistake would propagate to the caller code).

Reply:
Wording different - the developer who wrote the command should make the decision about "undoability" of the command itself, and the code using that command should not be able to modify that behavior.